### PR TITLE
Bug 1946079: Pass the API VIP to the downloader container

### PR DESCRIPTION
**Pull request by honza**
Depends on https://github.com/openshift/ironic-rhcos-downloader/pull/40

### CODE REVIEW

**Comment by sadasu** [comment]
@honza could you please add a little more information about the API VIP? As far as i know, this provides a way for remote workers not on the provisioning network to be able to download their image from the baremetal control plane (metal3 pod). Does not depend on the ProvisioningNetwork state.

**Comment by sadasu** [comment]
@Ian @asalkeld do we need to add any validation for this new config?

**Comment by hardys** [comment]
The API VIP is already stored in the platformStatus in the cluster infrastructure CR, can we use that instead of adding the provisioning CR field?

```
$ oc get infrastructures.config.openshift.io cluster -o yaml | yq .status.platformStatus
{
  "baremetal": {
    "apiServerInternalIP": "fd2e:6f44:5dd8:c956::5",
    "ingressIP": "fd2e:6f44:5dd8:c956::4"
  },
  "type": "BareMetal"
}
```

**Comment by stbenjam** [comment]
We have an epic to enable CBO on other platforms than baremetal, so I'm not sure relying on the baremetal platform in the Infrastructure CR is the right thing to do.

Also does the infrastructure CR exist outside of IPI  installs?

Could we do a DNS look-up instead? See if an A record exists, if not then see if there's an AAAA record?

File: provisioning/baremetal_pod_test.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func TestProxyAndCAInjection(t *testing.T) {
 	}{
 		{
 			name:       "init containers have proxy and CA information",
-			containers: newMetal3InitContainers(&images, managedProvisioning().build(), &proxy),
+			containers: newMetal3InitContainers(&images, managedProvisioning().build(), &proxy, ""),
 		},
 		{
 			name:       "metal3 containers have proxy and CA information",
```
Comment (sadasu) [inline provisioning/baremetal_pod_test.go:214]: Can we add a test where we pass in a valid API VIP and check if the container.Env for the "metal3-machine-os-downloader" contains this API VIP?

**Comment by asalkeld** [comment]
> We have an epic to enable CBO on other platforms than baremetal, so I'm not sure relying on the baremetal platform in the Infrastructure CR is the right thing to do. This will be empty on every other platform.
> 
> Also does the infrastructure CR exist outside of IPI installs?
> 
> Could we do a DNS look-up instead? See if an A record exists, if not then see if there's an AAAA record?

This is a really good point..
Note: that epic won't land this sprint, but we will have the same issue with capi cc @hardys

**Comment by honza** [comment]
Closing in favour of #148

**Pull request closed without merging**